### PR TITLE
[Merged by Bors] - chore: Rename Salem-Spencer sets to 3AP-free sets

### PR DESCRIPTION
**Pull request by YaelDillies**
Salem and Spencer were neither the first to write a paper on them, nor gave the best bounds, hence the current choice of name is a bit odd (and so is the Wikipedia page).

This PR renames `MulSalemSpencer` to `ThreeGPFree` and `AddSalemSpencer` to `ThreeAPFree`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/Additive/AP/Three.lean [hunk 1]
```diff
@@ -47,69 +51,72 @@ open Pointwise
...
-#align add_salem_spencer AddSalemSpencer
-
-/-- Whether a given finset is Salem-Spencer is decidable. -/
-@[to_additive "Whether a given finset is Salem-Spencer is decidable."]
-instance {α : Type*} [DecidableEq α] [Monoid α] {s : Finset α} :
-    Decidable (MulSalemSpencer (s : Set α)) :=
+/-- A set is **3GP-free** if it does not contain any non-trivial geometric progression of length
+three.
+
+This is also sometimes called a **non averaging set** or **Salem-Spencer set**. -/
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Additive/AP/Three.lean:65]: Is this true of the multiplicative version

File: Mathlib/Combinatorics/Additive/AP/Three.lean [hunk 2]
```diff
@@ -47,69 +51,72 @@ open Pointwise
...
+@[to_additive "A set is **3AP-free** if it does not contain any non-trivial geometric progression of
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Additive/AP/Three.lean:66]:
```suggestion
@[to_additive "A set is **3AP-free** if it does not contain any non-trivial arithmetic progression of
```

**Review by eric-wieser — APPROVED** [review]
Modulo docstrings errors the patch here looks fine; can you get another combinatorist to sign off on the rename?

File: Mathlib/Combinatorics/Additive/AP/Three.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Combinatorics/Additive/AP/Three.lean:65]: No, indeed

**Review by TFBloom — APPROVED** [review]
This is an appropriate change of terminology to agree with the literature.

**Review by eric-wieser — APPROVED** [review]
bors d+

One again, I'll let you decide if a deprecation for the definition is worthwhile.

File: Mathlib/Tactic/ToAdditive.lean [hunk 3]
```diff
@@ -1031,6 +1031,8 @@ def fixAbbreviation : List String → List String
                                       => "function" :: "_" :: "commute" :: fixAbbreviation s
   | "zero" :: "Le" :: "Part" :: s         => "posPart" :: fixAbbreviation s
   | "le" :: "Zero" :: "Part" :: s         => "negPart" :: fixAbbreviation s
+  | "three" :: "GPFree" :: s         => "three" :: "APFree" :: fixAbbreviation s
+  | "Three" :: "GPFree" :: s         => "Three" :: "APFree" :: fixAbbreviation s
```
Comment (fpvandoorn) [inline Mathlib/Tactic/ToAdditive.lean:1035]: Does `GPFree` or `gpfree` occur as a atomic part of any name that should not be translated to `APFree`/`apfree`? If there are no such cases, you can you just add `| "gpfree"=> ["apfree"]` to `nameDict`. That is the preferred way to translate "easy" cases.

**Pull request closed without merging**